packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,568 @@
+
+  @override
+  double computeDistanceToActualBaseline(TextBaseline baseline) {
+    return firstChild.getDistanceToActualBaseline(baseline);
+  }
+
+  @override
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! _SegmentedControlContainerBoxParentData) {
+      child.parentData = new _SegmentedControlContainerBoxParentData();
+    }
+  }
+
+  @override
+  void performLayout() {
+    double maxHeight = _kMinSegmentedControlHeight;
+    final Constraints parentConstraints = constraints.loosen();
+
+    int length = 0;
+    RenderBox child = firstChild;
+
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+      child.layout(parentConstraints, parentUsesSize: true);
+      final double boxHeight = child.size.height;
+      if (boxHeight > maxHeight) {
+        maxHeight = boxHeight;
+      }
+      child = childParentData.nextSibling;
+      length += 1;
+    }
+
+    final double overallWidth = constraints.maxWidth;
+    final Size overallSize = constraints.constrain(Size(overallWidth, maxHeight));
+
+    Offset centerLayout(Size boxSize, double x) {
+      assert(maxHeight >= boxSize.height);
+      final Offset boxOffset = new Offset(x, (maxHeight - boxSize.height) / 2.0);
+      return boxOffset;
+    }
+
+    final double childWidth = overallSize.width / length;
+
+    final BoxConstraints childConstraints =
+      new BoxConstraints.tightFor(width: childWidth, height: maxHeight);
+
+    child = firstChild;
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+      child.layout(childConstraints, parentUsesSize: true);
+      child = childParentData.nextSibling;
+    }
+
+    Offset childOffset = Offset.zero;
+
+    const double left = 0.0;
+    final double right = overallSize.width;
+
+    switch (textDirection) {
+      case TextDirection.rtl:
+        double start = right;
+        child = firstChild;
+        while (child != null) {
+          final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+          childOffset = centerLayout(Size(childWidth, maxHeight), start - childWidth);
+          childParentData.offset = childOffset;
+          final Rect childRect = Rect.fromLTWH(start - childWidth, 0.0, childWidth, maxHeight);
+          RRect roundedChildRect;
+          if (child == firstChild) {
+            roundedChildRect = RRect.fromRectAndCorners(childRect, topLeft: const Radius.circular(3.0),
+                bottomLeft: const Radius.circular(3.0));
+          } else if (child == lastChild) {
+            roundedChildRect = RRect.fromRectAndCorners(childRect, topRight: const Radius.circular(3.0),
+                bottomRight: const Radius.circular(3.0));
+          } else {
+            roundedChildRect = RRect.fromRectAndCorners(childRect);
+          }
+          _rects.add(roundedChildRect);
```

Comment:
instead of a separate _rects list, you could just store the rect on the parent data next to the offset.